classes/01class/exercises/c01-aws01/gpavelar/README.md

```diff
@@ -0,0 +1,436 @@
+
+```bash
+# Configure
+aws configure
+
+# Create key pair
+aws ec2 create-key-pair --key-name devops-class --query 'KeyMaterial' --output text > DevOps-academy.pem
+
+# Check my key
+aws ec2 describe-key-pairs --key-name devops-class --output table
+
+------------------------------------------------------------------------------------------------------------
+|                                             DescribeKeyPairs                                             |
++----------------------------------------------------------------------------------------------------------+
+||                                                KeyPairs                                                ||
+|+--------------------------------------------------------------+---------------+-------------------------+|
+||                        KeyFingerprint                        |    KeyName    |        KeyPairId        ||
+|+--------------------------------------------------------------+---------------+-------------------------+|
+||  0c:87:be:21:9d:aa:8d:45:d5:93:eb:11:3b:33:32:10:a8:fc:fa:c6 |  devops-class |  key-0a0f1ba7fa681e9fc  ||
+|+--------------------------------------------------------------+---------------+-------------------------+|
+# output
+sg-076a9212c8d1a8a45
+
+# Set the permissions of your private key
+chmod 400 DevOps-academy.pem
+
+# Create group
+aws ec2 create-security-group --group-name devops-develop --description "DevOps Development group" --vpc-id vpc-37d4dd50
+
+# Describe security groups
+aws ec2 describe-security-groups --group-names devops-develop
+
+# Command executed
+aws ec2 run-instances \
+  --image-id resolve:ssm:/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2 \
+  --count 1 \
+  --instance-type t2.micro \
+  --key-name devops-class \
+  --user-data file://devopsdata.txt \
+  --subnet-id subnet-351ced53 \
+  --security-group-ids sg-076a9212c8d1a8a45
+
+
+```
+
+Used config:
+
+- --image-id: ami-0e4bc04bd401729d6
+- --image-id: ami-09d95fab7fff3776c (From Website free tier eligible)
+- --image-id resolve:ssm:/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2
+- --count: 1
+- --instance-type: t2.micro
+- --key-name: devops-class
+- --user-data: file://devopsdata.txt
+- --subnet-id: subnet-351ced53
+- --security-group-ids sg-076a9212c8d1a8a45
+
+```bash
+
+# Authorize SSH
+# I performed the following command first and it didn't work...
+aws ec2 authorize-security-group-ingress --group-id sg-076a9212c8d1a8a45 --protocol tcp --port 22 --cidr 203.0.113.0/24
+
+# So, I performed this one
+aws ec2 authorize-security-group-ingress --group-id sg-076a9212c8d1a8a45 --protocol tcp --port 22 --cidr 0.0.0.0/0
+
+```
+
+I needed to learn a little bit about Classless Inter-Domain Routing, so here digged some question about that [here](https://stackoverflow.com/questions/44234374/aws-what-does-0-0-0-0-0-and-0-mean)
```

Comment:
What questions you have about it? Please ping us in the Slack channel.

In a nutshell, 0.0.0.0/0 means that all IP's are included in this range, from any network class.  So you use this CIDR block for allowing/denying traffic, routing and other purposes. 

take a look at this calculator to understand the first/last IPs and the size of the range: https://cidr.xyz/

Reply:
I changed this restriction limiting less IP's. So I removed all ingress rules and performed new ones.